bricks/dart_frog_prod_server/hooks/pre_gen.dart

```diff
@@ -21,11 +24,21 @@ Future<void> run(HookContext context) => preGen(context);
 Future<void> preGen(
   HookContext context, {
   io.Directory? directory,
+  ProcessRunner runProcess = io.Process.run,
   RouteConfigurationBuilder buildConfiguration = buildRouteConfiguration,
   void Function(int exitCode) exit = _defaultExit,
+  Future<void> Function(String from, String to) copyPath = io_expanded.copyPath,
 }) async {
   final projectDirectory = directory ?? io.Directory.current;
 
+  // We need to make sure that the pubspec.lock file is up to date
+  await dartPubGet(
```

Comment:
I think we should definitely allow the user to avoid performing the `dartPubGet`, maybe via a` --[no-]pub ` flag, similar to `flutter run --no--pub`. This flag would determine wether to run "dart pub get" before executing the generation, defaulting to on.

Reply:
I disagree, after my last commit, the brick now uses the pubspec lock file to get the external dependencies, since it provides us an accurate and reliable way to get all the external dependencies, even if they are external dependencies from external dependencies.

And like I added in the comment, we need to make sure that the pubspec.lock exists and is up to date.

Also, we already do a pub get in the generated bundle, with no option to skip it, because we do need the dependencies to be resolved to make the bundle, so I feel there is no real need for such flag.

Reply:
I feel the use case might apply in:
- Trying to use the command without internet connection
- Running on CI with a cached _pubspec.lock_

Reply:
We could, but I still find it dangerous. Let me exemplify why.

We could implement that flag, and do a validation, if you are passing the flag and there is no pubspec.lock, we give the user an error. That would cover the `Running on CI with a cached pubspec.lock` scenario that you mentioned.

But nothing guarantees that the pubspec.lock that is cached/checked under version control is up to date, and there is no way that we can check if it is (at least not without writing a quite complex code that would be potentially fragile) and give the user an reasonable error and then this can potentially break epically when bundling (this applies for the pub get that I introduced, as well to the pub get that is already being done at the end).

I still believe that this flag might introduce more danger, issues than benefits, so I would rather not consider adding it. But if feel otherwise, lets follow what you said and leave this for other PRs/versions :)